src/azure-cli-core/azure/cli/core/tests/test_keys.py

```diff
@@ -0,0 +1,120 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+import unittest
+import tempfile
+import paramiko
+import io
+import os
+import shutil
+
+from azure.cli.core.keys import generate_ssh_keys
+
+
+class TestGenerateSSHKeys(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        # set up temporary directory to be used for temp files.
+        cls._tempdirName = tempfile.mkdtemp(prefix="key_tmp_")
+
+        cls.key = paramiko.RSAKey.generate(2048)
+        keyOutput = io.StringIO()
+        cls.key.write_private_key(keyOutput)
+
+        cls.private_key = keyOutput.getvalue()
+        cls.public_key = '{} {}'.format(cls.key.get_name(), cls.key.get_base64())
+
+    @classmethod
+    def tearDownClass(cls):
+        # delete temporary directory to be used for temp files.
+        shutil.rmtree(cls._tempdirName)
+
+    def tearDown(self):
+
+        # remove all temporary files/directories created by previous test method.
+        dir = self._tempdirName
+        file_paths = [os.path.join(dir, name) for name in os.listdir(dir)]
+
+        for fp in file_paths:
+            if os.path.isfile(fp):
+                os.remove(fp)
+            else:
+                shutil.rmtree(fp)
```

Comment:
Way too complicated. One `setUp` `tearDown` pair and execute the same logic for each test method. There is no need to share the construction between test methods.